Fix:Remove tags from indicator

Prateek-Cyware opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Fixed remove tags from indicators command.

## Must have
- [x] Tests
- [x] Documentation

inbalapt1 commented on Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:
```diff
@@ -549,20 +549,19 @@ def tag_indicator_updation(
-        url_suffix = "ingestion/threat-data/action/add_tag/"
+            url_suffix = "ingestion/threat-data/action/remove_tag/"
+            tags = [_.strip() for _ in tag_id.split(",")]
+            data = {"tag_ids": list(set(tags))}
```
Why does the data here use the `tag_ids` field instead of `tag_id`, as in add_tag_indicator?

inbalapt1 commented on Packs/CTIX/ReleaseNotes/2_2_18.md:
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### CTIX v3
+- Fixed remove tags from indicators command.
```
```suggestion
- Fixed tag removal from indicators in the *ctix-add-tag-indicator* command.
```

inbalapt1 reviewed (commented):
Hi @Prateek-Cyware ,
Thanks for your contribution! I appreciate your work on this.

I noticed that in add_tag, the data includes all existing tags (including the current ones), whereas in remove_tag, only the tags to remove are sent. Is there a specific reason for this difference in behavior?

inbalapt1 commented:
Hi @Prateek-Cyware ,

During my internal tests, I noticed that the pre-commit is failing. Here are the following things that need to be fixed:
Packs/CTIX/Integrations/CTIXv3/CTIXv3.yml: [ST111] - Missing sectionorder key. Add sectionorder to the top of your YAML file and specify the order of the Collect, Connect, and Optimize sections (at least one is required).

Please see [here](https://xsoar.pan.dev/docs/integrations/yaml-file#configurations-sections) how to fix it.

In addition, please update the dockerimage tag, as described in here:
Packs/CTIX/Integrations/CTIXv3/CTIXv3.yml: [DO106] - docker image demisto/python3:3.11.10.115186's tag 3.11.10.115186 is outdated. The latest tag is 3.12.8.1983910

Prateek-Cyware commented:
> Hi @Prateek-Cyware ,
> 
> During my internal tests, I noticed that the pre-commit is failing. Here are the following things that need to be fixed: Packs/CTIX/Integrations/CTIXv3/CTIXv3.yml: [ST111] - Missing sectionorder key. Add sectionorder to the top of your YAML file and specify the order of the Collect, Connect, and Optimize sections (at least one is required).
> 
> Please see [here](https://xsoar.pan.dev/docs/integrations/yaml-file#configurations-sections) how to fix it.
> 
> In addition, please update the dockerimage tag, as described in here: Packs/CTIX/Integrations/CTIXv3/CTIXv3.yml: [DO106] - docker image demisto/python3:3.11.10.115186's tag 3.11.10.115186 is outdated. The latest tag is 3.12.8.1983910

I have pushed the changes as advised. thanks

inbalapt1 commented:
Hi @Prateek-Cyware, the code looks great! 

1. Please make sure to add a unit test for the add_tags_indicator operation case.
2. We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR. 
Alternatively, feel free to send me or add to the PR a recording of the demo.

Let me know if you need any help!

inbalapt1 commented:
> > Hi @Prateek-Cyware, the code looks great!
> > 
> > 1. Please make sure to add a unit test for the add_tags_indicator operation case.
> > 2. Please update the release notes to include the update of the docker image like this:
> >    `- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.`
> > 3. We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.
> >    Alternatively, feel free to send me or add to the PR a recording of the demo.
> > 
> > Let me know if you need any help!
> 
> Hi @inbalapt1 , I have updated the release notes , the unit test for add indicator tag command is already in place and i have made a little change in alignment to highlight that. I will update with demo video once it is ready. thanks

Hi @Prateek-Cyware, thanks!
I want to ask you please to go over these two pre-commit failures and fix them:
Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:559:37: E0606 Client.tag_indicator_updation: Possibly using variable 'url_suffix' before assignment
Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:564:20: E0606 Client.tag_indicator_updation: Possibly using variable 'data' before assignment

Prateek-Cyware commented:
> Hi @Prateek-Cyware, the code looks great!
> 
> 1. Please make sure to add a unit test for the add_tags_indicator operation case.
> 2. Please update the release notes to include the update of the docker image like this:
>    `- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.`
> 3. We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.
>    Alternatively, feel free to send me or add to the PR a recording of the demo.
> 
> Let me know if you need any help!

Hi @inbalapt1 , can you please let me know an email where I can send the demo or we can connect over call?

Prateek-Cyware commented:
> > > Hi @Prateek-Cyware, the code looks great!
> > > 
> > > 1. Please make sure to add a unit test for the add_tags_indicator operation case.
> > > 2. Please update the release notes to include the update of the docker image like this:
> > >    `- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.`
> > > 3. We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.
> > >    Alternatively, feel free to send me or add to the PR a recording of the demo.
> > > 
> > > Let me know if you need any help!
> > 
> > 
> > Hi @inbalapt1 , I have updated the release notes , the unit test for add indicator tag command is already in place and i have made a little change in alignment to highlight that. I will update with demo video once it is ready. thanks
> 
> Hi @Prateek-Cyware, thanks! I want to ask you please to go over these two pre-commit failures and fix them: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:559:37: E0606 Client.tag_indicator_updation: Possibly using variable 'url_suffix' before assignment Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:564:20: E0606 Client.tag_indicator_updation: Possibly using variable 'data' before assignment

Hi @inbalapt1 ,I have fixed the issue and it has passed the pre-commit.  can you please let me know an email where I can send the demo or we can connect over call?

inbalapt1 commented:
> > > > Hi @Prateek-Cyware, the code looks great!
> > > > 
> > > > 1. Please make sure to add a unit test for the add_tags_indicator operation case.
> > > > 2. Please update the release notes to include the update of the docker image like this:
> > > >    `- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.`
> > > > 3. We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.
> > > >    Alternatively, feel free to send me or add to the PR a recording of the demo.
> > > > 
> > > > Let me know if you need any help!
> > > 
> > > 
> > > Hi @inbalapt1 , I have updated the release notes , the unit test for add indicator tag command is already in place and i have made a little change in alignment to highlight that. I will update with demo video once it is ready. thanks
> > 
> > 
> > Hi @Prateek-Cyware, thanks! I want to ask you please to go over these two pre-commit failures and fix them: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:559:37: E0606 Client.tag_indicator_updation: Possibly using variable 'url_suffix' before assignment Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:564:20: E0606 Client.tag_indicator_updation: Possibly using variable 'data' before assignment
> 
> Hi @inbalapt1 ,I have fixed the issue and it has passed the pre-commit. can you please let me know an email where I can send the demo or we can connect over call?

Hi @Prateek-Cyware, you can send me the demo to this email - iapt@paloaltonetworks.com. thanks!

Prateek-Cyware commented:
> > > > > Hi @Prateek-Cyware, the code looks great!
> > > > > 
> > > > > 1. Please make sure to add a unit test for the add_tags_indicator operation case.
> > > > > 2. Please update the release notes to include the update of the docker image like this:
> > > > >    `- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.`
> > > > > 3. We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.
> > > > >    Alternatively, feel free to send me or add to the PR a recording of the demo.
> > > > > 
> > > > > Let me know if you need any help!
> > > > 
> > > > 
> > > > Hi @inbalapt1 , I have updated the release notes , the unit test for add indicator tag command is already in place and i have made a little change in alignment to highlight that. I will update with demo video once it is ready. thanks
> > > 
> > > 
> > > Hi @Prateek-Cyware, thanks! I want to ask you please to go over these two pre-commit failures and fix them: Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:559:37: E0606 Client.tag_indicator_updation: Possibly using variable 'url_suffix' before assignment Packs/CTIX/Integrations/CTIXv3/CTIXv3.py:564:20: E0606 Client.tag_indicator_updation: Possibly using variable 'data' before assignment
> > 
> > 
> > Hi @inbalapt1 ,I have fixed the issue and it has passed the pre-commit. can you please let me know an email where I can send the demo or we can connect over call?
> 
> Hi @Prateek-Cyware, you can send me the demo to this email - [iapt@paloaltonetworks.com](mailto:iapt@paloaltonetworks.com). thanks!

Hi @inbalapt1 , I have shared the demo over the email mentioned. thanks

Prateek-Cyware commented on Packs/CTIX/ReleaseNotes/2_2_18.md:
```diff
@@ -0,0 +1,4 @@
+#### Integrations
+##### CTIX v3
+-  Fixed tag removal from indicators in the *ctix-add-tag-indicator* command.
```
> 

I have pushed above  suggested change in release notes thanks.
